components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/internal/IdentitySAMLSSOServiceComponent.java

```diff
@@ -180,6 +182,10 @@ protected void activate(ComponentContext ctxt) {
         } finally {
             IdentityIOStreamUtils.closeInputStream(fis);
         }
+
+        ctxt.getBundleContext().registerService(AbstractEventHandler.class.getName(),
```

Comment:
Any reason why we don't do this within try block ? A throwable throwing from this registration may cause retries to activate the bundle again and again.

Reply:
fixed